kotlin-codegen/compiler/src/main/java/com/squareup/moshi/AdapterGenerator.kt

```diff
@@ -102,6 +104,14 @@ internal class AdapterGenerator(
 
   private fun generateType(): TypeSpec {
     val result = TypeSpec.classBuilder(adapterName)
+
+    generatedAnnotation(env.elementUtils, env.sourceVersion).ifPresent {
```

Comment:
the fact that this emits different things depending on what the env’s source version is, is disgusting. Can we just hardcode the `javax.annotation.Generated` one?

Reply:
That will prevent its use on Java 9 or newer.

Reply:
Of course, Moshi uses JSR 305 already so it's already prevented from being used on the module path so... that's actually not a terrible idea until we tear out JSR 305.

Reply:
Another advantage of using this is that it only emits if the annotation is available on the classpath which javax.annotation.Generated is not for Android projects.

Reply:
Can’t we use this?
 https://docs.oracle.com/javase/9/docs/api/javax/annotation/Generated.html

Reply:
You can, but it's in a super weird module that must be explicitly depended on and the entire module is marked as `@Deprecated(forRemoval=true)` so eventually this will need changed. And then you still need to conditionally generate for Android.

Reply:
Some details in https://github.com/google/auto/issues/560

Reply:
I dislike the premise that you get a different annotation based on what JDK you’re compiling with.

Can we just skip `@Generated`? Seems like the juice isn’t worth the squeeze.

Reply:
That sounds simple!

Reply:
That is simpler, but this is also useful for static analysis tools to know it's generated code. I agree it's a bit awkward due to the change in java 9, but it's also inline with what Dagger, AutoValue, etc all do. I'd definitely want to have some way to at least optionally enable it, even if off by default. What do you think?

Reply:
If we support anything it should be opt-in via one of the annotation processor args: `-Amoshi.generated=javax.annotation.Generated`

If we add this we should discourage its use because it prevents a single class from being used in multiple environments. Everyone's life is easier if their generated code works on all platforms: Android, Java 8, Java 9.

Reply:
Done in 436f36f

Also reverted the elements -> processingEnv change in b454aaa

For the life of me I can't figure out how to specify apt options in maven kapt though :|